src/transformers/file_utils.py

```diff
@@ -84,31 +84,23 @@
 if USE_TF in ENV_VARS_TRUE_AND_AUTO_VALUES and USE_TORCH not in ENV_VARS_TRUE_VALUES:
     _tf_available = importlib.util.find_spec("tensorflow") is not None
     if _tf_available:
+        candidates = (
+            "tensorflow",
+            "tensorflow-cpu",
+            "tensorflow-gpu",
+            "tf-nightly",
+            "tf-nightly-cpu",
+            "tf-nightly-gpu",
+            "intel-tensorflow",
+        )
         # For the metadata, we have to look for both tensorflow and tensorflow-cpu
-        try:
-            _tf_version = importlib_metadata.version("tensorflow")
-        except importlib_metadata.PackageNotFoundError:
+        for pkg in candidates:
             try:
-                _tf_version = importlib_metadata.version("tensorflow-cpu")
+                _tf_version = importlib_metadata.version(pkg)
+                break
             except importlib_metadata.PackageNotFoundError:
-                try:
-                    _tf_version = importlib_metadata.version("tensorflow-gpu")
-                except importlib_metadata.PackageNotFoundError:
-                    try:
-                        _tf_version = importlib_metadata.version("tf-nightly")
-                    except importlib_metadata.PackageNotFoundError:
-                        try:
-                            _tf_version = importlib_metadata.version("tf-nightly-cpu")
-                        except importlib_metadata.PackageNotFoundError:
-                            try:
-                                _tf_version = importlib_metadata.version("tf-nightly-gpu")
-                            except importlib_metadata.PackageNotFoundError:
-                                # Support for intel-tensorflow version
-                                try:
-                                    _tf_version = importlib_metadata.version("intel-tensorflow")
-                                except importlib_metadata.PackageNotFoundError:
-                                    _tf_version = None
-                                    _tf_available = False
+                pass
+        _tf_available = _tf_version is not None
```

Comment:
I'm confused on how this does not throw an error if tensorflow is not available (it apparently does not since the tests are passing): `_tf_version` is never initialized to `None` so there should be an error when TF is not installed (about `_tf_version` not being defined).

For my peace of mind, can you add a `_tf_version = None` before the loop?

Reply:
It doesn't throw because if one doesn't have any version of TensorFlow installed, we never enter the 
```py
    if _tf_available:
```
statement!

This would crash if we were to install another version of TensorFlow that is not in the list of candidates. One that comes to mind is the TensorFlow fork for M1 macs: https://github.com/apple/tensorflow_macos

Reply:
Thank you for catching it, @sgugger!, indeed `_tf_version` would be undefined if all candidates fell through, but not the general "tensorflow". Fixed!